### PR TITLE
feat: Added leaderboard track selection dropdown

### PR DESCRIPTION
**Pull request by Panzerhandschuh**
Closes momentum-mod/game/issues/2116

### CODE REVIEW

File: scripts/pages/leaderboards/leaderboards.js [hunk 1]
```diff
@@ -173,4 +176,43 @@ class Leaderboards {
...
+				text: trackStr,
+				value: index
+			});
+			item.SetAttributeInt('trackNum', leaderboard.trackNum);
+			item.SetAttributeInt('trackType', leaderboard.trackType);
+
+			this.panels.tracksDropdown.AddOption(item);
+
+			index++;
+		}
```
Comment (tsa96) [inline scripts/pages/leaderboards/leaderboards.js:208]: If you use array methods you don't need to handle `index` yourself here, also may as well only call `GetCurrentGamemode` once:
```js
const gamemode = GamemodeAPI.GetCurrentGameMode();
map.leaderboards
  .filter(leaderboard => leaderboard.gamemode == gamemode)
  .forEach((leaderboard, index) => { 
      ...
   })
```
could also redestructure the `leaderboards` in the forEach to `{trackType, trackNum}`.

**Review by tsa96 — CHANGES_REQUESTED** [review]

File: scripts/pages/leaderboards/leaderboards.js (hunk 1 above)
Comment (Panzerhandschuh) [reply scripts/pages/leaderboards/leaderboards.js:208]: I'm not sure what you mean by restructuring the leaderboards into trackType/num. Can you give me a code snippet for that too?

Comment (tsa96) [reply scripts/pages/leaderboards/leaderboards.js:208]: Sorry, I wrote "redestructure" when I meant "destructure"!

Here's how I'd write the whole thing:
```js
const currentMode = GamemodeAPI.GetCurrentGameMode();
map.leaderboards
  .filter(({ gamemode }) => gamemode === currentMode)
  .forEach(({ trackType, trackNum }, index) => {
    let trackStr;
    if (trackType === 0) {
      trackStr = "Main Track";
    } else if (trackType === 1) {
      trackStr = `Stage ${trackNum + 1}`;
    } else {
      trackStr = `Bonus ${trackNum + 1}`;
    }

    const item = $.CreatePanel("Label", this.panels.tracksDropdown, trackStr, {
      text: trackStr,
      value: index
    });
    item.SetAttributeInt("trackNum", trackNum);
    item.SetAttributeInt("trackType", trackType);

    this.panels.tracksDropdown.AddOption(item);
  });
  ```
I don't think destructuring is totally necessary here btw, it's really a stylistic preference and I wouldn't have brought it up if I wasn't mentioning the other stuff.

**Comment by Panzerhandschuh** [comment]
Don't approve this yet. I forgot that I still need to add localization.

**Review by tsa96 — APPROVED** [review]

**Pull request merged**